### ISSUE TITLE
Update process

### ISSUE THREAD

**Issue by exo1987** [openhab/openhab-qnap-qpkg#37]
HI  guys,

is it necessary for the upgrade/update process to remove the openHAB qnap app and also delete the usersdata folder before installing a new version? Is there no easier way?

Kind regards
Exo

**Comment by Andy1402** [comment]
I am not sure if there is an easier way. I always tried the safe way: Safe all configs, delete the old app and install the new qpgk. Thenafter I copied all configs into the fresh installation.

**Comment by exo1987** [comment]
HI Andy1402, thank you for your answer. Do you mean with all configs the config folder or other folders, too? I usually delete all folders after deinstallation and then copy just the config folder to the new installation. It take a long time to install all bindings and do the configuration thats why I was asking about  an other way...

**Comment by Andy1402** [comment]
Hi exo1987.
I mean I copied everything in my conf folder. If you have sth. in you addon folder, save it, too. Then I deinstalled the app. During the deinstallation all files will be deleted. And then I installed the new version from the QPKG and restored the conf files, manually. Maybe there is an easier way, but I don't know it it really works. I never tried it.

The bindings are installed automatically in my case because of my addons.cfg. You can configure, that bindings are automatically installed. In my case:
binding = exec,fritzboxtr0641,http1,km1001,network,ntp,samsungtv,zwave
ui = basic,paper,habmin,classic,habpanel
persistence = mysql
action = mail,telegram
So I don't need to install them step by step manually...
Regards.
Andy

**Comment by exo1987** [comment]
HI Andy, thats a very good tip! I was able to find a file with the name addons.config (not .cfg) but it seems like it is the same you were talking about. This will safe a lot of time....

regards
Exo

**Comment by 6r15u** [comment]
@Andy1402 did you backup userdata folder? oder the folders within that folder?
I'm quite new to openHAB, and did not configure much until now as I first wanted to get the setup straight.
Do you know what is necessary and what can be left out? I've seen that in userdata sometimes are configs which lead to issues when updating?

**Comment by Andy1402** [comment]
@6r15u no I did a backup from the folder "conf" only.

**Comment by thopiekar** [comment]
@Andy1402 That is too less. Most of openHAB's settings are stored in a database, which is again stored in userdata. Therefore, if you only backup conf, then you only save your smarthomes' rules and other general settings. As a consequence you'll need readd all devices and configure them again. (in case you lost your settings)

**Comment by Andy1402** [comment]
@thopiekar you are right. But in my case it is enough. I'm using a MySQL DB for database. And yupp, I configure my Z-Wave-Stick manual first time and add every Z-Wave-Thing again. I'm not sure how to prevent this. In the past I read from problems, if you simply overwrite the userdata folder.

**Comment by thopiekar** [comment]
Well, it is a tricky part. What you can do is making a backup of your working userdata directory and make an update, so you get the new directory. Then you need to copy all files from your backup to this one. Optionally if you are unlucky there will be old remains in the backup, which need to be removed.
Already did it like that and it saved half of the day of reconfiguring everything.
